tad/api/routes/projects.py

```diff
@@ -0,0 +1,63 @@
+import logging
+from typing import Annotated
+
+from fastapi import APIRouter, Depends, Request
+from fastapi.responses import HTMLResponse
+
+from tad.api.deps import templates
+from tad.core.config import get_settings
+from tad.core.exceptions import UnsafeFileError
+from tad.schema.instrument import InstrumentBase
+from tad.schema.project import ProjectNew
+from tad.schema.system_card import SystemCard
+from tad.services.instruments import InstrumentsService
+from tad.services.projects import ProjectsService
+from tad.services.storage import WriterFactory
+from tad.utils.path import is_safe_path
+
+router = APIRouter()
+logger = logging.getLogger(__name__)
+
+
+@router.get("/new")
+async def get_new(
+    request: Request,
+    instrument_service: Annotated[
+        InstrumentsService, Depends(InstrumentsService)
+    ],  # todo: create async_instruments_service
+) -> HTMLResponse:
+    instruments = await instrument_service.fetch_instruments()
+
+    return templates.TemplateResponse(request, "projects/new.html.j2", {"instruments": instruments})
+
+
+@router.post("/new", response_class=HTMLResponse)
+async def post_new(
+    request: Request,
+    project_new: ProjectNew,
+    projects_service: Annotated[ProjectsService, Depends(ProjectsService)],
+) -> HTMLResponse:
+    system_card_file = get_settings().CARD_DIR / f"{project_new.name}_system.yaml"
+    if not is_safe_path(get_settings().CARD_DIR, system_card_file):
+        raise UnsafeFileError(system_card_file)
+
+    model_card_file = get_settings().CARD_DIR / f"{project_new.name}_model.yaml"
+    if not is_safe_path(get_settings().CARD_DIR, model_card_file):
+        raise UnsafeFileError(model_card_file)
+
+    project = projects_service.create(project_new.name, system_card_file)
+
+    instruments: list[InstrumentBase] = [
+        InstrumentBase(urn=instrument_urn) for instrument_urn in project_new.instruments
+    ]
+    system_card = SystemCard(name=project_new.name, selected_instruments=instruments)
+
+    storage_writer = WriterFactory.get_writer(
+        writer_type="file", location=system_card_file.parent, filename=system_card_file.name
+    )
+    storage_writer.write(system_card.model_dump())
+
+    # TODO(Berry): Parse instruments and add them to the project
+
+    headers = {"HX-Location": f"/project/{project.id}"}
+    return templates.TemplateResponse(request, "redirect.html.j2", headers=headers)
```

Comment:
We could abstract this maybe into a "redirect response", with an interface like 
```python
templates.Redirect(request, f"/project/{project.id}")
```

Reply:
Nice idea